gradle/testing/defaults-tests.gradle

```diff
@@ -181,6 +181,13 @@ allprojects {
         // Print some diagnostics about locations used.
         logger.info("Test folders for {}: cwd={}, tmp={}", project.path, testsCwd, testsTmpDir)
       }
+      if(isCIBuild) {
+        retry {
+          maxRetries = 20
+          maxFailures = 20
```

Comment:
I read the docs on these two.
maxFailures -- maybe 10 instead?  If there are 11 failures or more, I'd think something is broadly wrong and we shouldn't do retries.  
maxRetries -- 20 seems like a lot; do we really want to rerun a test as many as 20 times?  I'd say 5 is plenty.

Reply:
Ohh did I commit `maxRetries = 20`. No It should not be that high, I believe I have committed the setting that I was testing for the randomization functionality. I think `maxRetries = 3` is alright.

Reply:
What I initially thought was this?

The maxFailure setting acts as a cumulative failure limit. If it's set to 10, and you have 5 test cases  fail, in the worst case scenario where retries also fail, it might retry the first 3 test cases 3 times, the 4th one only once, and skip the 5th one since the maxFailure limit would be reached. And I have seen around 4-5 test cases failing the recent build. I thought It would be great to give all the test cases a chance. With 20, we could re-run, in worst case, 7 tests. 

But I am also fine with 10 as well.

Reply:
You're correct. The maxRetries setting applies to each round, and if the total failures in that round exceed the limit, no retries are attempted. I retested with multiple tests, and this confirmed it, clearing up any doubts. I think 10 is perfect then!